### PR TITLE
Feature/impact hub

### PR DESCRIPTION
**Pull request by dodo721**

### CODE REVIEW

**Comment by dodo721** [comment]
This was meant to be on master anyway - but figured I'd make it a pull request.
Entirely bug fixes

File: src/js/components/LandingSection.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ const LandingSection = (): JSX.Element => {
 		<>
 			{/* 
 			//@ts-ignore */}
-			<MyLoopNavBar logo='/img/logo-in-pill.svg' backgroundColor='transparent' />
+			<MyLoopNavBar logo="/img/new-logo-with-text-white.svg" logoScroll="/img/new-logo-with-text.svg"/>
```
Comment (winterstein) [inline src/js/components/LandingSection.tsx:35]: This is fine as-is. FYI, /img/gl-logo gets you to a set of canonical logos, from the shared https://github.com/good-loop/icon repo

File: src/js/components/ShareButton.jsx [hunk 2]
```diff
@@ -9,19 +17,42 @@ class ShareButton extends React.Component {
...
+		let paramCount = 0;
+		params.forEach (param => {
+			url += (paramCount != 0 ? "&" : "") + param;
+			paramCount ++;
+		});
+		url += (paramCount != 0 ? "&" : "") + "link=" + escape(this.props.url);
+		console.log("Sharing: " + url);
+		url = escape(url);
+		console.log("Escaped: " + url);
+
```
Comment (winterstein) [inline src/js/components/ShareButton.jsx:44]: Hm - Let's make a util function in miscutils.ts for url building. Or does modern js have one built in? @roscoemcinerney?

File: src/js/components/ShareButton.jsx [hunk 3]
```diff
@@ -9,19 +17,42 @@ class ShareButton extends React.Component {
...
+	
 	render () {
+
+		// Generate ShareServlet sharing url
+		let metaProps = this.props.meta;
+		if (!metaProps)
+			metaProps = this.props;
+		let params = [];
+		if (metaProps.title)
+			params.push('title=' + escape(metaProps.title));
```
Comment (winterstein) [inline src/js/components/ShareButton.jsx:29]: Minor point: please use our `encURI` over `escape`, as it supports unicode.

File: src/js/components/ShareButton.jsx (hunk 2 above)
Comment (winterstein) [reply src/js/components/ShareButton.jsx:44]: https://stackoverflow.com/questions/486896/adding-a-parameter-to-the-url-with-javascript/44160941#44160941

```
let url = new URL("http://foo.bar/");
url.searchParams.set('x', 42);
url.toString();
```

Comment (winterstein) [reply src/js/components/ShareButton.jsx:44]: That's not supported in IE, so best practice would be to write a utility function in miscutils.ts. That way we can put in an IE shim if we need one (without having to dig up every usage of searchParams).

Also the utility function could take an object of key-values and set them all.

Comment (winterstein) [reply src/js/components/ShareButton.jsx:44]: Actually: I just tested URLSearchParams. It encodes space as `+`, which should be considered unsafe (not all decoding functions will decode that).

So: best AVOID searchParams. And let's definitely write our own utility method.

Comment (dodo721) [reply src/js/components/ShareButton.jsx:44]: Done a quick google and found JS does have URL building utils. I'll swap them out

File: src/js/components/ShareButton.jsx (hunk 3 above)
Comment (dodo721) [reply src/js/components/ShareButton.jsx:29]: Will do!

**Comment by dodo721** [comment]
Just pushed some advert changes ive been working on with the URL change by accident - will finish them before the merge

File: src/js/components/LandingSection.tsx (hunk 1 above)
Comment (dodo721) [reply src/js/components/LandingSection.tsx:35]: Where is the gl-logo repo desginated to go - ~/winterwell/icon/gl-logo?

**Pull request merged**